olive/systems/azureml/aml_pass_runner.py

```diff
@@ -103,12 +102,24 @@ def main(raw_args=None):
     # this is to handle passes like OrtPerfTuning that use the same model file as input
     model_json["same_model_path_as_input"] = False
     if model_json["config"]["model_path"] is not None:
-        model_path = str(Path(model_json["config"]["model_path"]).resolve())
-        if model_path == input_model_path:
+        # create a resource path from the model path
+        model_resource_path = ResourcePath.create_resource_path(model_json["config"]["model_path"])
+        # string representation of the model path
+        model_path_str = model_resource_path.get_path()
+        if model_path_str == input_model_path:
+            # if the model path is the same as the input model path, set model_path to None
+            # and set same_model_path_as_input to True
             model_json["config"]["model_path"] = None
             model_json["same_model_path_as_input"] = True
-        else:
-            model_json["config"]["model_path"] = str(Path(model_path).relative_to(Path(common_args.pipeline_output)))
+        elif model_resource_path.type in [ResourceType.LocalFile, ResourceType.LocalFolder]:
```

Comment:
I am supposing the model_resource_path should be always local path here?

Reply:
It can also be a `ResourceType.StringName`

Reply:
Currently, all our passes create model with local files/folders so the StringName case doesn't happen. But I think it is still better to cover all cases?

Reply:
I could not imagine which case could generate the output model with only string names.

Reply:
Done. asserts that it is local as discussed offline